src/standalone/Vegetation/canopy_energy.jl

```diff
@@ -29,3 +37,99 @@ temperature.
 """
 canopy_temperature(model::PrescribedCanopyTempModel, canopy, Y, p, t) =
     canopy.atmos.T(t)
+
+## Prognostic Canopy Temperature
+"""
+    BigLeafEnergyParameters{FT <: AbstractFloat}
+"""
+struct BigLeafEnergyParameters{FT <: AbstractFloat}
+    ρc_canopy::FT
+end
+
+function BigLeafEnergyParameters{FT}(; ρc_canopy = FT(2e6)) where {FT}
+    return BigLeafEnergyParameters{FT}(ρc_canopy)
+end
+
+
+"""
+    BigLeafEnergyModel{FT} <: AbstractCanopyEnergyModel{FT}
+"""
+struct BigLeafEnergyModel{FT} <: AbstractCanopyEnergyModel{FT}
+    parameters::BigLeafEnergyParameters{FT}
+end
+
+ClimaLSM.prognostic_vars(model::BigLeafEnergyModel) = (:T,)
+ClimaLSM.prognostic_types(model::BigLeafEnergyModel{FT}) where {FT} = (FT,)
+ClimaLSM.prognostic_domain_names(model::BigLeafEnergyModel) = (:surface,)
+
+"""
+    canopy_temperature(model::BigLeafEnergyModel, canopy, Y, p, t)
+
+Returns the canopy temperature under the `BigLeafEnergyModel` model, 
+where the canopy temperature is modeled prognostically.
+"""
+canopy_temperature(model::BigLeafEnergyModel, canopy, Y, p, t) =
+    Y.canopy.energy.T
+
+function make_compute_exp_tendency(model::BigLeafEnergyModel, canopy)
+    function compute_exp_tendency!(dY, Y, p, t)
+        area_index = p.canopy.hydraulics.area_index
+        FT = eltype(t)
+        h_c = canopy.hydraulics.compartment_surfaces[end]
+        ρc_canopy = model.parameters.ρc_canopy
+
+        net_energy_flux = @. -p.canopy.radiative_transfer.LW_n -
```

Comment:
@braghiere here is where the equation is set for change in canopy T. rhoc_canopy = volumetric heat capacity (rho * specific heat capacity). ai = above ground area index. h_c = canopy height. 

the energy_roots is the energy carried by the water in root extraction and is needed to conserve energy since this is energy that the soil loses. All fluxes are positive if pointing from smaller depth to larger depth (positive = towards atmos, or towards canopy from soil), except the LW and SW net radiation.